### PR TITLE
Fixes #2248: Updating plop templates to generate components with dynamic filenames

### LINKED ISSUE #2248: Custom plop action "renameMany" is not required
# 📝 Feedback

Creating a custom plop action "renameMany" to have dynamic filenames is not required.  Dynamic file-naming can be done by renaming template filenames appropriately. Please refer to the code sample for an example.

This can significantly reduce the amount of code.

## 🔦 Context

To my understanding, the plop action renameMany is being used to rename the generated files to include the component's name in the filename. This can be handled with only handlebars and plop, and without the overhead of using a custom action.

In this way, we can totally remove a custom plop action simple change in template filenames.

## 💻 Code Sample/Alternative
For example, if the filename `Component.tsx.hbs` is changed to `{{componentName}}.tsx.hbs`, the generated file will automatically have the variable componentName within the filename.

```
<!-- If you have an example are seeing an error, please provide a code repository, gist or sample files to reproduce the issue -->

## 🌍 Your Environment

<!--- Include as many relevant details about the environment you experienced the issue in -->

| Software         | Version(s) |
| ---------------- | ---------- |
| [Package name]   |
| Browser          |
| Operating System |

## 🧢 Your Company/Team

<!--- Which product team is requesting this feedback? (i.e. Adobe/Photoshop) -->

### PR DESCRIPTION
**Pull request by neeraj3029**
Closes #2248 

Current workflow: 
- plop generates components 
- plop action `renameMany.js` is used to rename the generated files, to include the component name 

What this PR brings:
- Plop now puts the component name directly in the templates. Hence, now the extra step of using plop-action renameMany becomes obsolete.

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x]  Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

- Run `yarn run plop` and answer prompts
- Observe the generated files

### CODE REVIEW

**Comment by snowystinger** [comment]
Thanks! you'll need to sign the CLA, then once you've done that, close and reopen the PR
https://react-spectrum.adobe.com/contribute.html#contributor-license-agreement

File: scripts/plopfile.js [hunk 1]
```diff
@@ -15,8 +15,7 @@ let rspProject = 'React Spectrum v3';
 let otherProject = 'other';
 
 module.exports = function (plop) {
-  plop.setActionType('renameMany', require('../plop-actions/renameMany'));
-
+
```
Comment (snowystinger) [inline scripts/plopfile.js:18]: if there's no more references to renameMany, we should delete it in this PR as well

Comment (neeraj3029) [reply scripts/plopfile.js:18]: Deleted the renameMany.js file and signed the CLA :)

**Review by snowystinger — APPROVED** [review]
Tested locally, looks good, thanks for doing this

**Review by dannify — APPROVED** [review]

**Pull request merged**